content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/execute.rb

```diff
@@ -16,6 +16,7 @@ def main
                 @handle.log("info", "Starting execute")
 
                 begin
+                  service&.set_automate_timeout(@handle.field_timeout, service_action)
```

Comment:
@lfu The safe navigation is not doing anything here.  If the `service` object is nil it will raise an error below in the `def service` method.  Not sure if you are trying to protect against the service object not exposing the `set_automate_timeout`, if so this will not help.